subsys/bluetooth/host/gatt.c

```diff
@@ -4234,6 +4230,11 @@ int bt_gatt_unsubscribe(struct bt_conn *conn,
 
 	params->value = 0x0000;
 
+	/* Don't attempt to write if not connected */
+	if (conn->state != BT_CONN_CONNECTED) {
```

Comment:
I'm wondering if the unsubscribe should be queued, in the same way as a service changed would be. 
Unsubscribing this way now would mean that the peer would never be informed, and could just keep sending notifications that would be silently dropped.

Reply:
We would need to add memory storage for unsubscribing it so the argument would be then if it is worth the extra memory, also note that we do have a similar problem with BT_GATT_SUBSCRIBE_FLAG_VOLATILE as it does unsubscribe automatically as well, since this is on the client side I don't think it breaks anything but perhaps you want it to be stricly used with unpair?

Reply:
What's the point of unsubscribing without letting the peer know? The app could simply drop the notifications instead once it reconnects?